src/query/functions-v2/src/scalars/string_multi_args.rs

```diff
@@ -309,4 +240,106 @@ pub fn register(registry: &mut FunctionRegistry) {
             }),
         }))
     });
+
+    registry.register_function_factory("char", |_, args_type| {
+        if args_type.is_empty() {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "char",
+                args_type: vec![DataType::UInt64; args_type.len()],
+                return_type: DataType::String,
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_, _| None),
+            eval: Box::new(char_fn),
+        }))
+    });
+
+    // nullable char
+    registry.register_function_factory("char", |_, args_type| {
+        if args_type.is_empty() {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "char",
+                args_type: vec![DataType::Nullable(Box::new(DataType::UInt64)); args_type.len()],
+                return_type: DataType::Nullable(Box::new(DataType::String)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_, _| None),
+            eval: Box::new(wrap_nullable(char_fn)),
+        }))
+    });
+}
+
+fn concat_fn(args: &[ValueRef<AnyType>], _: &GenericMap) -> Result<Value<AnyType>, String> {
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+    let args = args
+        .iter()
+        .map(|arg| arg.try_downcast::<StringType>().unwrap())
+        .collect::<Vec<_>>();
+
+    let size = len.unwrap_or(1);
+    let mut builder = StringColumnBuilder::with_capacity(size, 0);
+    for idx in 0..size {
+        for arg in &args {
+            unsafe { builder.put_slice(arg.index_unchecked(idx)) }
+        }
+        builder.commit_row();
+    }
+
+    match len {
+        Some(_) => Ok(Value::Column(Column::String(builder.build()))),
+        _ => Ok(Value::Scalar(Scalar::String(builder.build_scalar()))),
+    }
+}
+
+fn char_fn(args: &[ValueRef<AnyType>], _: &GenericMap) -> Result<Value<AnyType>, String> {
+    let args = args
+        .iter()
+        .map(|arg| arg.try_downcast::<UInt64Type>().unwrap())
+        .collect::<Vec<_>>();
+
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+    let input_rows = len.unwrap_or(1);
+
+    let mut values: Vec<u8> = vec![0; input_rows * args.len()];
+    let values_ptr = values.as_mut_ptr();
+
+    for (i, arg) in args.iter().enumerate() {
```

Comment:
What if `v` is not in the range of `u8`? For example: `select char(77, 111111, 77);`. MySQL will return `NULL` if there is any invalid value(arg). And MySQL will ignore `NULL` in values(args).

Reply:
I think we can just accept u8 as the signature. For other behaviors in MySQL, we don't need to follow strictly.